app/scripts/comp/util/date-format.js

```diff
@@ -0,0 +1,68 @@
+import { SettingsManager } from 'comp/settings/settings-manager';
+
+const DateFormat = {
+    getMonthsForLocale() {
+        const format = new Intl.DateTimeFormat(SettingsManager.activeLocale, { month: 'long' });
+        const months = [];
+        for (let month = 0; month < 12; month++) {
+            months.push(format.format(new Date(Date.UTC(2008, month))));
+        }
+        return months;
+    },
+
+    getWeekDaysForLocale() {
+        const format = new Intl.DateTimeFormat(SettingsManager.activeLocale, { weekday: 'long' });
+        const weekdays = [];
+        for (let day = 1; day < 8; day++) {
+            weekdays.push(format.format(new Date(Date.UTC(2007, 9, 6 + day))));
+        }
+        return weekdays;
+    },
+
+    getWeekdaysShortForLocale() {
+        const format = new Intl.DateTimeFormat(SettingsManager.activeLocale, { weekday: 'short' });
+        const weekdays = [];
+        for (let day = 1; day < 8; day++) {
+            weekdays.push(format.format(new Date(Date.UTC(2007, 9, 6 + day))));
+        }
+        return weekdays;
+    },
+
+    dtStr(dt) {
+        if (typeof dt === 'number') {
+            dt = new Date(dt);
+        }
+        return dt
+            ? new Intl.DateTimeFormat(SettingsManager.activeLocale, {
+                  year: 'numeric',
+                  month: 'short',
+                  day: 'numeric',
+                  hour: '2-digit',
+                  minute: '2-digit',
+                  second: '2-digit'
+              }).format(dt)
```

Comment:
I've fixed tests, but this is a bit not correct in `en-US`. It returns `03 AM` and `13` without PM. You can fix this if you want, or I'll do it later with other fixes, I don't have time to do it today already.

Reply:
May I ask, why did you choose this way of specifying format, and not `timeStyle: "medium"`? I'm trying to understand the issue that you were trying to solve.

Reply:
The previous code used to pad values in order to have always 2 digit, so i wanted to make it works as expected (always padded);
`timeStyle: "medium"`allows value like 4:00 AM instead of 04:00 AM, that's why i used this specific '2-digit'.

Reply:
I see, thanks! It should be padded according to locale rules, so I’ll replace it with short or medium then.